### PR TITLE
Feature test

### PR DESCRIPTION
**Pull request by ronitsawant00**
## Summary by Sourcery

Introduce core service implementations, CLI applications, and automated reviews.

New Features:
- Add TaskService and TaskService2 implementations for CRUD operations on Task entities
- Introduce Python expense tracker CLI script with add, total, and category filtering features
- Implement Java product management console application with ProductManager and Main entry point

CI:
- Add GitHub Actions workflow for Sourcery code analysis on pull request diffs

Tests:
- Add comprehensive JUnit 5 test suites for TaskService and TaskService2 covering all CRUD operations and edge cases

## Summary by CodeRabbit

- **New Features**
  - Task management APIs: list, fetch, create, partial-update, delete, and filter tasks by status.
  - Console Product Manager and Product data model: view all, add, and find products.
  - JSON-backed Expense Tracker CLI: add expenses, view total, filter by category.
- **Bug Fixes**
  - Clearer "not found" errors for tasks.
  - Prevented overwriting fields with empty/null values during task updates.
- **Tests**
  - Extensive unit tests covering task service behaviors, partial updates, deletion, and status filtering.
- **Chores**
  - Added a PR-check GitHub Actions workflow.

### CODE REVIEW

**Comment by ronitsawant00** [comment]
@sourcery-ai review

**Comment by ronitsawant00** [comment]
sourcery review --fix

**Comment by ronitsawant00** [comment]
@sourcery-ai review --fix

**Comment by ronitsawant00** [comment]
@sourcery-ai resolve

File: expensetrcker.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if e["category"]==cat:
+            filtered.append(e)
+    return filtered
+
+
+if __name__=="__main__":
+    while True:
+        print("1.Add Expense\n2.View Total\n3.View by Category\n4.Exit")
+        ch=int(input("Enter choice:"))
+        if ch==1:
```
[the remark this answers is not in the document]
Comment (ronitsawant00) [reply expensetrcker.py]: Resolved

**Comment by ronitsawant00** [comment]
@sourcery-ai review --fix
